src/model/quotes.js

```diff
@@ -131,14 +122,21 @@ class QuotesModel {
     }
     if (interceptQuoteEvents.length > 0) {
       // send the quote request to the recipient in the event
-      return {
+      const result = {
         terminate: false,
         quoteRequest,
         headers: {
           ...headers,
           'fspiop-destination': interceptQuoteEvents[0].params.rerouteToFsp
         }
       }
+      // if rerouteToFspCurrency is present then add the non standard headers used by forex
+      // Note these headers are not part of the mojaloop specification
+      if (interceptQuoteEvents[0].params.rerouteToFspCurrency) {
```

Comment:
As discussed, please re-factor this to be generic, and rather have the rules provide "additionalHeaders" which can be injected.

Reply:
At the very least, this type of functionality should be behind a feature switch.

Reply:
@lewisdaly that was also discussed.